### DISCUSSION TITLE
how to prevent module hoisting in workspace?

### DISCUSSION THREAD

**Discussion by ellemedit** [oven-sh/bun#10382] [Q&A] [answered]
I'm managing couple of next.js and Vite React projects with Bun workspace. Each project has different next.js/react versions.

Bun version is `1.1.4`.

In this situation.

```
<project root>
packages/
  a/
  b/
```

If `a` and `b` both have `next` dependency. It resolved like:

```
<project root>
node_modules/
  next(for `a`)
packages/
  a/
  b/
    node_modules/
      next(for `b`)
```

I'd expected:

```
<project root>
packages/
  a/
    node_modules/
      next
  b/
    node_modules/
      next
```

Currently, I do this by adding temporal dependency like:

```
<project root>
node_modules/
  next (for `a-temporal-hoisting-prevent-module`)
pakcages/
  a-temporal-hoisting-prevent-module/
  a/
    next
  b/
    next
```

This is little confusion and flaky.

Is there better way to resolve this? Preventing hoist is important for next.js as they are peering react deps with different versions.

**Comment by vinkovsky** [comment]
did you fix the problem?

**Comment by PaulRBerg** [comment, marked as answer]
You can now avoid hoisting by adding this setting to your `bunfig.toml`:

```toml
[install]
  linker = "isolated"
```

Docs: https://bun.com/docs/install/isolated
